core/arch/arm/kernel/generic_core_bootcfg.c

```diff
@@ -149,6 +146,9 @@ static struct map_area bootcfg_memory_map[] = {
 	{
 	 .type = DEVICE1_TYPE,
 	 .pa = DEVICE1_BASE, .size = DEVICE1_SIZE,
+#ifdef DEVICE1_VA_BASE
```

Comment:
Would prefer:
* rename `DEVICE1_BASE` as `DEVICE1_PA_BASE`
* Remove `#ifdef DEVICE1_VA_BASE`. That means we should always define `DEVICE1_VA_BASE`, defining it to 0 when 1-to-1 mapping is requested.